controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +171,14 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		if err := r.reconcileCephRadosNamespace(storageCluster, consumerConfig); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephFilesystemSubVolumeGroup(storageCluster, consumerConfig); err != nil {
+			return reconcile.Result{}, err
+		}
+
```

Comment:
You need to have a guard here that check for configmap controller ship before invoking these resources.
That guard should also encapsulate all of the other steps that are in this scope